model/userdata.go

```diff
@@ -22,14 +25,13 @@ const (
 
 // UserData represents userdata which might be split across multiple storage types
 type UserData struct {
-	Parts map[string]*UserDataPart
+	Parts            map[string]*UserDataPart
+	templateFilePath string
 }
 
 type UserDataPart struct {
-	Asset    Asset
-	tmpl     *template.Template
```

Comment:
keeping original template allows UserDataPart to be lazy which is a requirement for `extraContext` calls in CF stack template